plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go

```diff
@@ -229,6 +229,13 @@ func ClusterRoles() []rbac.ClusterRole {
 				rbac.NewRule("create").Groups(authorizationGroup).Resources("subjectaccessreviews").RuleOrDie(),
 			},
 		},
+		{
+			// a role to use for the kube-dns pod
+			ObjectMeta: api.ObjectMeta{Name: "system:kube-dns"},
+			Rules: []rbac.PolicyRule{
+				rbac.NewRule("list", "watch").Groups(legacyGroup).Resources("endpoints", "services").RuleOrDie(),
```

Comment:
> Would kube-dns have the permission to fetch ConfigMap in general with this setup? Fetching ConfigMap might not be a right permission for kube-dns but we introduced it (#36775) as a short term fix for the federation issue.

Like I've set it up here, no.  I saw that access and assumed it was a bug.  This role is a good starting point and we can have a separate pull to add questionable permissions.